### ISSUE TITLE
namespaces tag names not matched

### ISSUE THREAD

**Issue by johanneswilm** [dperini/nwsapi#5]
@dperini This looks great, but the selector issues for XML seem to be the as in nwmatcher.

This works::

```js
let parser = new window.DOMParser()
let string = `<?xml version="1.0"?><cp:coreProperties xmlns:cp="http://schemas.openxmlformats.org/package/2006/metadata/core-properties" xmlns:dc="http://purl.org/dc/elements/1.1/"><dc:title></dc:title></cp:coreProperties>`
let dom = parser.parseFromString(string, 'text/xml')
let el = dom.querySelector('coreProperties')
```

But if one replaces the querySelector with the one from nwsapi, it does not::

```js
NW.Dom.install()
let parser = new window.DOMParser()
let string = `<?xml version="1.0"?><cp:coreProperties xmlns:cp="http://schemas.openxmlformats.org/package/2006/metadata/core-properties" xmlns:dc="http://purl.org/dc/elements/1.1/"><dc:title></dc:title></cp:coreProperties>`
let dom = parser.parseFromString(string, 'text/xml')
let el = dom.querySelector('coreProperties')
```

What does however work is::

```js
dom.querySelector('cp\\:coreProperties')
```

Which in turn does not work with the native querySelector found in browsers. 

Same behavior as in nwmatcher seen in https://github.com/jsdom/jsdom/issues/2159

**Comment by dperini** [comment]
@johanneswilm I am going to check this out.
Could you provide more XML namespace testing ?
That could help identifying and possibly correct the issue you spotted.
Unfortunately I do not have exhaustive XML test suite, also I am trying to find and add some.

Thank you for reporting the issue.

**Comment by johanneswilm** [comment]
I can try to find some more, although the selectors that I am aware of that have XML namespace are really just the element selector where the namespace is left out as mentioned in this issue:

```js
dom.querySelector('coreProperties') // matches <cp:coreProperties>
```

And selectors with namespaced attributes:

```js
dom.querySelector('[*|href]') // matches <svg xlink:href="foo"/>
dom.querySelector('svg[*|href]') // matches <svg xlink:href="foo"/>
dom.querySelector('[*|href="foo"]') // matches <svg xlink:href="foo"/>
dom.querySelector('[*|w="0"][*|type="dxa"]') // matches <w:tblInd w:w="0" w:type="dxa"/>
```
This last issue was also reported here: https://github.com/jsdom/jsdom/issues/2028

**Comment by dperini** [comment]
This is a first attempt to fix this issue as requested in dperini/nwsapi#5. This is related to missing support to handle XML namespaced tag names selectors.
Please check that it fixes the issue that you reported !

Thank you for the report, much appreciated.

**Comment by johanneswilm** [comment]
Hey, thanks for acting so fast!

This now works:

```js
dom.querySelector('coreProperties') // matches <cp:coreProperties>
```

The attribute selectors still do not seem to work as they do in the native implementation, such as:

```js
let parser = new window.DOMParser()
let string = `<?xml version="1.0"?><cp:coreProperties xmlns:cp="http://schemas.openxmlformats.org/package/2006/metadata/core-properties" xmlns:dc="http://purl.org/dc/elements/1.1/"><dc:title dc:something="hey"></dc:title></cp:coreProperties>`
let dom = parser.parseFromString(string, 'text/xml')
dom.querySelector('[*|something]')
```

This does work in the native implementations of Firefox and Chrome.

**Comment by dperini** [comment]
Landed another patch in 
https://github.com/dperini/nwsapi/commit/bd336aa76d2004da5d95d480dcb4503358cf5b91 for this same problem, I removed unnecessary leftover code calling in 'lookupNamespaceURI' which breaks in node/jsdom.

**Comment by johanneswilm** [comment]
Hmm... still does not seem to work for me. I'm running it in the browser with `NW.Dom.install()`.

**Comment by johanneswilm** [comment]
Ok, it looks like it's working now!

**Comment by dperini** [comment]
I found that the "SlickSpecs" test suite I recently added contains a lot of XML tests. Even if it is a bit dated it perfectly fits the needs. You can run these test and see that there are also many test related to SVG elements in HTML documents.

**Comment by johanneswilm** [comment]
I saw that. Good idea. Several ways css selectors for xml are special that I was not aware of.

**Comment by dperini** [comment]
These latest commits should have solved also https://github.com/jsdom/jsdom/issues/2028

**Comment by willstott101** [comment]
It doesn't look like it's worked...

I've created a small demo here: https://jsfiddle.net/willstott101/8596evs2/

Which shows that in browsers, all tagName queries ignore namespaces unless the query is prefixed with `|`.

I believe this behaviour can be fixed by a patch SIMILAR to https://github.com/willstott101/nwsapi/commit/42e881eedbc94a49cb9fb552c53095f84336767f - the problem with my commit here, is that it doesn't support the `|` prefix.

So to summarise:
* Namespaced elements are not yet working.
* We can support them by using `getElementsByTagNameNS("*", name)` instead of `getElementsByTagName(name)`.
* We still want to use `getElementsByTagName(name)` if the tagName in the query is prefixed with `|`

I don't think it would be too complicated to fix, but I find the source and tests in this repo a bit hard to navigate, I'll try to open a PR with my progress.

**Comment by willstott101** [comment]
I've opened a PR - the test I've added does make it look like the patch worked... though to be honest IDK why the `|` prefix seems to query correctly.
